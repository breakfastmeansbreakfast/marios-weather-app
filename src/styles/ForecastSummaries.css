.forecast-summaries {
    display: flex;
    justify-content: space-between;
}

.forecast-summary__icon {
    height: 75px;
    margin-bottom: 10px;
}