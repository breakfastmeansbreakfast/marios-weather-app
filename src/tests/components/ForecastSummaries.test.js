import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = {
    forecasts: [
      {
        date: 1111111,
        description: "Stub description 1",
        icon: "stubIcon1",
        temperature: {
          max: 22,
          min: 12,
        },
      },
      {
        date: 2222222,
        description: "Stub description 2",
        icon: "stubIcon2",
        temperature: {
          max: 24,
          min: 13,
        },
      },
    ],
    onForecastSelect: () => {},
  };

  it("renders correctly", () => {
    const { asFragment } = render(
    <ForecastSummaries
      forecasts={validProps.forecasts}
      onForecastSelect={validProps.onForecastSelect} 
    />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps.forecasts} />
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  });
});
