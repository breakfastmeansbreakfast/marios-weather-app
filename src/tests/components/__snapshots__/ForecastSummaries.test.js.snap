// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        1111111
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 1
      </div>
      <div
        class="forecast-summary__icon"
      >
        stubIcon1
      </div>
      <div
        class="forecast-summary__temperature"
      >
        22°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 1
      </div>
    </div>
    <div
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        2222222
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 2
      </div>
      <div
        class="forecast-summary__icon"
      >
        stubIcon2
      </div>
      <div
        class="forecast-summary__temperature"
      >
        24°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
